crates/red_knot_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -391,6 +405,45 @@ impl<'db> UseDefMap<'db> {
             .is_always_false()
     }
 
+    pub(crate) fn is_attribute_assignment_visible(
+        &self,
+        db: &dyn crate::Db,
+        binding: &BindingWithConstraints<'_, 'db>,
+        class_body_scope: ScopeId,
+    ) -> Truthiness {
+        let definition = binding.binding.as_ref().unwrap();
+        // The attribute assignment inherits the visibility of the method which contains it
+        let maybe_method_scope = definition.scope(db);
+        let is_method_visible = if let Some(method_def) = maybe_method_scope.node(db).as_function()
```

Comment:
Do we have any tests for an attribute assignment inside a definitely-not-existing method? I.e. do any tests fail if we remove this `is_method_visible` check?

Reply:
I added a test to check the visibility of an attribute defined in a definitely-non-existing-method.